tests/src/test/scala/BinaryTreeTest.scala

```diff
@@ -0,0 +1,235 @@
+/**
+ * Created by Nicolas A Perez (@anicolaspp) on 1/29/16.
+ */
+
+package dogs
+package tests
+
+import dogs.Order.Ordering
+import dogs.Predef._
+import org.scalatest._
+import tests.utils._
+
+class BinaryTreeTestAdd extends FlatSpec with Matchers {
+
+   implicit val cmp =  new Order[Int] {
+    override def apply(l: Int, r: Int): Ordering = l.compareTo(r) match {
+      case 0  => Order.EQ
+      case -1 => Order.LT
+      case 1  => Order.GT
+    }
+  }
+
+  "Binary Tree" should "be a leaft when empty" in {
+    val emptyTree = BinaryTree.empty[Int]
+
+    emptyTree should be(BTNil())
+  }
+
+  it  should "return a branch with one item when add to empty tree" in {
+
+    val empty = BinaryTree.empty[Int]
+
+    val branch = empty.add(5)
+
+    branch.value should be(Some(5))
+  }
+
+  it should "insert value into left branch when add to brach to the left" in {
+
+    val branch = BinaryTree(5)
+
+    val result = branch.add(3)
+
+    result.value should be(Some(5))
+    result.left.value should be(Some(3))
+  }
```

Comment:
I'm not sure why we care if it is Branch(3, _, Branch(5, _, _) vs. Branch(5, Branch(3, _, _), _)